ui/jest.config.js

```diff
@@ -33,5 +33,5 @@ module.exports = {
   ],
   coverageDirectory: './coverage',
   coverageReporters: ['html', 'cobertura'],
-  reporters: ['default', 'jest-junit'],
+  reporters: ['default', ['jest-junit', { classNameTemplate: '{filepath}' }]],
```

Comment:
Before this change, the JUnit reports generated by `jest` used the test name (free text full of spaces) as the class name. This isn't a perfect replacement, but it's enough to make Circle happy.

Reply:
We are assuming we're careful not to have spaces in our filepath, which seems reasonable. Only savages do that.